Homework 3

Tabolich-Aliaksei opened the pull request.

Tabolich-Aliaksei committed:
Create task_3_1

Tabolich-Aliaksei committed:
Create task_3_2

Tabolich-Aliaksei committed:
Create task_3_3

Tabolich-Aliaksei committed:
Create task_3_4

DmitryChitalov commented on task_3_1:
```diff
@@ -0,0 +1,140 @@
+        dct.popitem()           #O(1)
+
+
+pop_dict(my_dict)
```
не прочитано ДЗ
" сделайте аналитику, что заполняется быстрее и почему
   сделайте замеры времени"

DmitryChitalov commented on task_3_2:
```diff
@@ -0,0 +1,46 @@
+        print('Не верный пароль!')
+
+
+func_pass()
```
passwor_verification = input('Enter your password, please: ')
    pass_verif_hash = hashlib.sha256(salt.encode() + passwor_verification.encode()).hexdigest()


дублирование
нужно вынести в отд. ф-цию
посмотрите пример

DmitryChitalov commented on task_3_3:
```diff
@@ -0,0 +1,29 @@
+res_hash = set([(hashlib.sha256(f'{s[x: y]}'.encode('utf-8'))).hexdigest() for x in range(len(s)) for y in range(x + 1, len(s) + 1)])
+
+print(res)
+print(res_hash)
```
выполнено

DmitryChitalov commented on task_3_4:
```diff
@@ -0,0 +1,28 @@
+        cash[url] = (hashlib.sha512(salt.encode() + url.encode()).hexdigest())
+
+
+hash_cash_url('www.gb.ru')
```
выполнено

Tabolich-Aliaksei committed:
bug fix

for i in my_list[0:3]: # O(n)
изменение сложности на константную
for i in my_list[0:3]: # O(1)
